instruqt-tracks/terraform-cloud-aws-v2/02-terraform-cloud-setup/check-workstation

```diff
@@ -19,14 +36,85 @@ cd /root/hashicat-aws
+cd /root/terraform-cloud
+sed -i "s/YOUR_ORGANIZATION/$ORG/g" terraform.tfvars # This was originally looking in terraform.tfvars.example
+sed -i "s/YOUR_WORKSPACE/$WORKSPACE/g" terraform.tfvars
+# mv terraform.tfvars.example terraform.tfvars
+# The above command will get rid of the terraform.tfvars.example file, but when
+# we run the check script more than once, it cannot find that file bc it no longer
+# exists. However, to keep the consistency between code blocks, we kept this the 
+# same as above where we search and replace in the remote_backend.tf file. 
 
 # Store the ORG in /root/.bashrc
 grep $ORG /root/.bashrc || echo "export ORG=\"$ORG\"" >> /root/.bashrc
 
 # Store the WORKSPACE in /root/.bashrc
 grep $WORKSPACE /root/.bashrc || echo "export WORKSPACE=\"$WORKSPACE\"" >> /root/.bashrc
 
-exit 0
+# Do we have a valid token
+if [ -z "${TFC_TOKEN}" ]; then
+    fail-message "Unable to find Terraform Cloud Token, please double-check the \"3- Terraform Cloud Token\" steps"
+    exit 1
+fi
+
+STATUS=$(curl \
+    --header "Authorization: Bearer ${TFC_TOKEN}" \
+    --header "Content-Type: application/vnd.api+json" \
+    --request GET \
+    -w "%{response_code}" \
+    -s \
+    -o /tmp/.out.json \
+    https://app.terraform.io/api/v2/account/details 2>/dev/null)
+
+if [ "${STATUS}" != "200" ]; then
+    echo "Failed to get account details, status ${STATUS}"
+    if [ -f /tmp/.out.json ]; then
+	echo "Output was:"
+	cat /tmp/.out.json
+	echo "End of output"
+    fi
+    fail-message "Terraform Cloud Token is not valid, please double-check the \"3- Terraform Cloud Token\" steps"
+    exit 1
+fi
+
+rm -f /tmp/.out.json
+echo "We have a valid TFC token"
+
+# Is it a trial organization
+STATUS=$(curl \
+    --header "Authorization: Bearer ${TFC_TOKEN}" \
+    --header "Content-Type: application/vnd.api+json" \
+    --request GET \
+    -w "%{response_code}" \
+    -s \
+    -o /tmp/.out.json \
+    https://app.terraform.io/api/v2/organizations/"${ORG}" 2>/dev/null) # originally TFC_ORG
+
+if [ "${STATUS}" != "200" ]; then
+    echo "Failed to get organization entitlements, status ${STATUS}"
+    if [ -f /tmp/.out.json ]; then
+	echo "Output was:"
+	cat /tmp/.out.json
+	echo "End of output"
+    fi
+    fail-message "Unable to get TFC Organization Entitlements, please go back to the \"2- Terraform Cloud Trial Plan\" steps"
+    exit 1
+fi
+
+TRIAL=$(jq -r '.data.attributes."plan-is-trial"' /tmp/.out.json 2>/dev/null)
+if [ -z "${TRIAL}" ]; then
+    echo ".data.attributes.plan-is-trial was empty"
+    fail-message "Unable to determine TFC Trial Status, please go back to the \"2- Terraform Cloud Trial Plan\" steps"
+    exit 1
+fi
+
+if [ "${TRIAL}" != "true" ]; then
+    echo ".data.attributes.plan-is-trial was '${TRIAL}' not 'true'"
+    fail-message "Your TFC Organization is not a trial one, please go back to the \"2- Terraform Cloud Trial Plan\" step"
+    exit 1
+fi
+
+exit 0
```

Comment:
My main concern with just looking for `plan-is-trial: true` is that we don't really know what the future might bring as far as trial and pricing changes.  The `entitlement-sets` endpoint (`https://app.terraform.io/api/v2/organizations/[org-name]/entitlement-set`) will tell us granularly exactly what entitlements somebody has for their org.  Ideally we would look for the entire set of entitlements that go into the T&G trial entitlement set but for purposes of this check I think all we really need to check is the `data.attributes.versioned-policy-set-limit` key -- if it's `null`, then there is no limit and we should be free to proceed; if it is `0` then the org was not created with the right entitlements.

Reply:
I'll note that `plan-is-trial: true` worked for what I needed it for in the HVD work I'm doing, but if someone can enumerate what exact set of entitlements we do need for this track, I'd be fine with that as well. Someone needs to figure that out now, however, and test it.

Reply:
I *tried* figuring out what `?trial=workshop2023` did in terms of entitlements but was absolutely unable to do so. If we know who to ask that might be a good start, since that was added explicitly to make these workshops work again.